### DISCUSSION TITLE
Question on style training

### DISCUSSION THREAD

**Discussion by ManglerFTW** [TheLastBen/fast-stable-diffusion#1155] [Q&A] [answered]
I am wanting to train a model that can produce a variety of characters at a high quality. For instance, I want to train a model that can output high detail images of different types of armor or costumes say for a rogue, knight, wizard, etc. But I ALSO want to train the same model on an environment style. 

All the documentation I can find only shows how to train a single subject so it's not clear to me how to set this up. My first test method was to keep adding instance images with the file names being a prompt instead of x(1).jpg, x(2).jpg ... But I am now finding that as I continue to add instance and concept images in this way, I need to keep raising the CFG scale to get coherent results. 

How do I set up instances to train on multiple subjects? Say with my model keyword being x I want to train a model on knights and I add 20 images of different knights as instance images labeled as x(1).jpg, x(2).jpg ..., then I want to resume training at a later time, but on wizards. Do I add 20 additional instance images of different wizards as x(3).jpg, x(4).jpg... or would it be better to name the first set as knight(1).jpg, knight(2).jpg... and the second set as wizard(1).jpg, wizard(2).jpg...? If I do rename it as "knight"/"wizard", will that make it so that instead of having to put "x" in the prompt, you can put "knight" or "wizard" and it will generate an image based on the instances and concepts labeled as "wizard"?

**Comment by TheLastBen** [comment]
> or would it be better to name the first set as knight(1).jpg, knight(2).jpg... and the second set as wizard(1).jpg, wizard(2).jpg...

Yes correct, but don't use known words, use rare identifiers like "kndfgontr" and "oiyekivne" so you can summon the trained subjects with those identifiers.

upload all the images together, don't use concept images for now.

**Comment by ManglerFTW** [comment, marked as answer]
Thank you for your speedy reply! I have retrained it using the method you provided and I can see what you mean.

If you don't mind, I have some follow up questions. What purpose would the concept images serve in training using this method? Is it possible to build from this? As in say I trained on Knights and Wizards, but then I decided I wanted to add Aliens and in building the Alien dataset I found some cool new pictures of knights or wizards and wanted to add them in as well, but keep the original training. Is that possible? 

Could you resume training with new instances like x1rAliens (1).jpg, x1rAliens (2).jpg... + x1rKnight (21).jpg, x1rKnight (22).jpg..., x1rWizard (21).jpg, etc.?

**Reply by TheLastBen** [reply]
Yes you can resume with the previous instance images, but if they are not from the same class the new subject, you can simply train on only the new subject to avoid overtraining the previous subjects.

once you have done the training, if you notice a lack in diversity, say in knights, you can upload random pictures of knights as concept images and resume the training setting everything to 0 except the concept steps, set them to 500-1000 and train on the concept to widen the range of the instance "knight".

**Reply by ManglerFTW** [reply]
Awesome! Thank you so much. This clears up a lot. Time to run a few tests. :)

**Comment by Zaskek** [comment]
Adding my 2 cents based on some fantasy models i tried to train:make sure to add both environmental and class specific (class here meaning rogue, archer, etc... not the meaning of class in machine learning) in the instance images. Also try to aim to have some close ups of faces in the style you want there since otherwise you'd get really bad ones. Fullbody pics might seem as a fine idea to fill the instances with but i found that it gets the body right even without them, so you can get a little bit more detail on the body and armor while using the model. Anything above medium full shots is good. Make sure to have some fullbody pics tho, otherwise it doesnt get the boots and foot armor right. Depending on what you are using as a base model to finetune you might not need concept images or you might need a loooot of them, still experimenting with this aspect. 
Also, get prepared to see a lot of eldricht horror wielding weapons, tech still isn't here to properly draw a knight holding a straight claymore with two hands it seems lol.
Good luck, post some results!

**Reply by ManglerFTW** [reply]
I mean... she IS holding it with two hands. Close enough right? 

![04588-2362491546-CHV3CWizard, CHV3MTroll, full body shot, photo of the most beautiful artwork in the world featuring a morbidly obese (woman) in](https://user-images.githubusercontent.com/113069374/210303797-03372171-4d75-4176-9d22-bfe62fbcc6e0.png)
